manifests/osfamily/debian.pp

```diff
@@ -7,7 +7,7 @@
 
     if getvar('::puppet_agent::is_pe') == true {
       $pe_server_version = pe_build_version()
-      $source = "https://${::servername}:8140/packages/${pe_server_version}/${::platform_tag}"
+      $source = "https://${::puppet_master_server}:8140/packages/${pe_server_version}/${::platform_tag}"
```

Comment:
I'm not certain that `puppet_master_server` will contain the right value when a node has been configured with `server_list` only (and no `server` setting)

Reply:
So... I tested it and yes... if you specify a `server_list` setting without specifying a `server`, it'll set the fact to the default setting of `puppet`.  Good catch.

Two paths forward that I can think of:

1. Update the fact to use the value for `server` _or_ the first element in the `server_list` array if it is defined.
2. Scrap the whole thing and just do a README update.

We could just update the default value for `$source` as it have it in this PR anyway, but that could potentially be a breaking change for the few people out there that do use `server_list`.

Reply:
Option 1 presented in new commit c48427d (compliments to @Sharpie).